server/routes/profilesRoutes.js

```diff
@@ -1,43 +1,91 @@
-const mongoose = require("mongoose");
-const Profile = mongoose.model("profiles");
+const { ObjectId } = require("mongodb");
 
-const profileRoutes = (app) => {
-  app.get(`/api/profile`, async (req, res) => {
-    const profiles = await Profile.find();
+/**
+ * @param {import('express').Express} app - The Express instance
+ * @param {import('mongodb').Db} db - The Db instance.
+ */
+const profilesRoutes = (app, db) => {
+  /**
+   * Retrieves the profiles collection from Mongo db
+   * @returns Collection<Document>
+   */
+  const profilesCollection = () => db.collection("profiles");
 
-    return res.status(200).send(profiles);
+  /**
+   * Middleware handler for GET requests to /api/profiles path
+   */
+  app.get(`/api/profiles`, async (req, res) => {
+    try {
+      // Waits for asynchronous `find()` operation to complete and converts results to array
+      const profiles = await profilesCollection().find({}).toArray();
+
+      return res.status(200).send(profiles);
+    } catch (e) {
+      return res
+        .status(500)
+        .send(`Error occurred while retrieving profiles: ${e}`);
+    }
   });
 
-  app.post(`/api/profile`, async (req, res) => {
-    const profile = await Profile.create(req.body);
+  /**
+   * Middleware handler for POST requests to /api/profiles path
+   */
+  app.post(`/api/profiles`, async (req, res) => {
+    try {
+      const profile = await profilesCollection().insertOne(req.body);
 
-    return res.status(201).send({
-      error: false,
-      profile,
-    });
+      return res.status(201).send({
+        error: false,
+        profile,
+      });
+    } catch (e) {
```

Comment:
Could this be updated to be error instead of e? Will help reinforce that we mean error and not something else such as event